### ISSUE TITLE
Ilustration transform with original coco keypoint label

### ISSUE THREAD

**Issue by jinfagang** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#30]
Hi, this work is very good. However I got some puzzels about this:

```
for i in range(len(annotation['keypoints']) // 3):
                keypoint = [annotation['keypoints'][i * 3], annotation['keypoints'][i * 3 + 1], 2]
                if annotation['keypoints'][i * 3 + 2] == 1:
                    keypoint[2] = 0
                elif annotation['keypoints'][i * 3 + 2] == 2:
                    keypoint[2] = 1
                keypoints.append(keypoint)
            prepared_annotation['keypoints'] = keypoints
```
What does this transform do compare to original coco data label format?

**Comment by fy-meng** [comment]
The COCO keypoint format is in the form of `kpt_0_x, kpt_0_y, kpt_0_vis, kpt_1_x, kpt_1_y, kpt_1_vis, ...`, where `kpt_i_vis` is 0 if not labeled, 1 if not visible and 2 if visible. (http://cocodataset.org/#format-data)
I think the code here is just converting `kpt_i_vis` into a boolean value representing the visibility of the corresponding keypoint.

**Comment by jinfagang** [comment]
@fy-meng  thanks, sounds something like:

```
coco: [23, 34, 0, 45, 56, 1, 65, 89, 1, 90, 80, 2...]
->
 [23, 34, False, 45, 56, False, 65, 89, False, 90, 80, True...]
```
？ which converts 0 and 1 to False and 2 to True? ([x,y,v...] x, y is the point)

**Comment by Daniil-Osokin** [comment]
This code sets visibility for keypoints. If visibility is zero, we won't generate a peak in ground truth heatmap in training. So, as @fy-meng mentioned, visibility of all keypoints which are not labeled (value 0) or not visible (value 1) in COCO labels will be zero.

**Comment by Daniil-Osokin** [comment]
Hope, it is clear now.
